tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/api/rest/v1/controllers/VariablesController.java

```diff
@@ -52,7 +57,11 @@ public class VariablesController extends ApiErrorController {
   @GetMapping("{variableId}")
   public ResponseEntity<VariableResponse> getVariableById(
       @PathVariable @Parameter(description = "The ID of the variable.", required = true)
-          String variableId) {
+          final String variableId) {
+    if (!tasklistPermissionServices.hasPermissionToReadProcessDefinition(WILDCARD_RESOURCE)) {
```

Comment:
Should we really check `Read` on `ProcessDefinition` here? Or rather `ReadUserTask` or `ReadProcessInstance` on `ProcessDefinition` (note that the latter is currently not migrated from Management Identity)?

Reasoning being that people won't expect `Read` on `ProcessDefinition` to expose any process instance data. They might assign this permission to someone to see the models but not confidential runtime data such as a customer's process payload. This endpoint could leak them then.